stm32cube/CMakeLists.txt

```diff
@@ -51,6 +51,11 @@ if(CONFIG_CLOCK_STM32_HSE_CLOCK)
   zephyr_compile_definitions( -DHSE_VALUE=${CONFIG_CLOCK_STM32_HSE_CLOCK} )
 endif()
 
+if(CONFIG_PTP_CLOCK_STM32_HAL)
+  zephyr_compile_definitions_ifdef(CONFIG_ETH_STM32_HAL_API_V2
+    -DHAL_ETH_USE_PTP)
+endif()
```

Comment:
Nit: I'd make it the other way round:
```
if(CONFIG_ETH_STM32_HAL_API_V2)
  zephyr_compile_definitions_ifdef(CONFIG_PTP_CLOCK_STM32_HAL
    -DHAL_ETH_USE_PTP)
endif()
```
1. This is more in line with other changes
2. Once new API is fully validated and the previous one is deprecated, we just need to remove the outer if() condition